osu.Game/Online/Metadata/OnlineMetadataClient_Friends.cs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR.Client;
+
+namespace osu.Game.Online.Metadata
+{
+    public partial class OnlineMetadataClient
```

Comment:
Similarly, here instead of partials maybe we can get some composition / layering going. Something like

```
MetadataClient.UserPresence.UpdateStatus()
MetadataClient.DailyChallenge.DailyChallengeUpdated()
MetadataClient.Friends.OnFriendConnected()
```

etc.

I'm not necessarily a partial hater, but also I would prefer that things didn't devolve into complete grab-bags...